bundles/org.openhab.binding.revogi/src/main/java/org/openhab/binding/revogi/internal/RevogiSmartStripControlHandler.java

```diff
@@ -0,0 +1,174 @@
+import org.openhab.core.thing.Thing;
+import org.openhab.core.thing.ThingStatus;
+import org.openhab.core.thing.ThingStatusDetail;
+import org.openhab.core.thing.binding.BaseThingHandler;
+import org.openhab.core.types.Command;
+import org.openhab.core.types.RefreshType;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * The {@link RevogiSmartStripControlHandler} is responsible for handling commands, which are
+ * sent to one of the channels.
+ *
+ * @author Andi Bräu - Initial contribution
+ */
+@NonNullByDefault
+public class RevogiSmartStripControlHandler extends BaseThingHandler {
+
+    private final Logger logger = LoggerFactory.getLogger(RevogiSmartStripControlHandler.class);
+    private final StatusService statusService;
+    private final SwitchService switchService;
+    private @Nullable ScheduledFuture<?> pollingJob;
+
+    private @Nullable RevogiSmartStripControlConfiguration config;
+
+    public RevogiSmartStripControlHandler(Thing thing) {
+        super(thing);
+        UdpSenderService udpSenderService = new UdpSenderService(new DatagramSocketWrapper(), scheduler);
+        this.statusService = new StatusService(udpSenderService);
+        this.switchService = new SwitchService(udpSenderService);
+    }
+
+    @Override
+    public void handleCommand(ChannelUID channelUID, Command command) {
+        switch (channelUID.getId()) {
+            case RevogiSmartStripControlBindingConstants.PLUG_1_SWITCH:
+                switchPlug(command, 1);
+                break;
+            case RevogiSmartStripControlBindingConstants.PLUG_2_SWITCH:
+                switchPlug(command, 2);
+                break;
+            case RevogiSmartStripControlBindingConstants.PLUG_3_SWITCH:
+                switchPlug(command, 3);
+                break;
+            case RevogiSmartStripControlBindingConstants.PLUG_4_SWITCH:
+                switchPlug(command, 4);
+                break;
+            case RevogiSmartStripControlBindingConstants.PLUG_5_SWITCH:
+                switchPlug(command, 5);
+                break;
+            case RevogiSmartStripControlBindingConstants.PLUG_6_SWITCH:
+                switchPlug(command, 6);
+                break;
+            case RevogiSmartStripControlBindingConstants.ALL_PLUGS:
+                switchPlug(command, 0);
+                break;
+            default:
+                logger.debug("Something went wrong, we've got a message for {}", channelUID.getId());
+        }
+    }
+
+    private void switchPlug(Command command, int port) {
+        @Nullable
+        RevogiSmartStripControlConfiguration localConfig = this.config;
+        if (localConfig == null) {
+            logger.warn("No config available, config object was null");
+            return;
+        }
+        if (command instanceof OnOffType) {
+            int state = convertOnOffTypeToState(command);
+            switchService.switchPort(localConfig.getSerialNumber(), localConfig.ipAddress, port, state);
```

Comment:
Would be good to make the access to the config parameters consistent. You could remove the getter and access the fields directly.

Reply:
hm, in my code the usage is consistent. Everywhere else the properties are accressed directly?

Reply:
In the first argument of `switchPort()` a getter is used and in the second argument, the field is accessed directly.